test/e2e/framework/util.go

```diff
@@ -5130,3 +5130,16 @@ func CreateFileForGoBinData(gobindataPath, outputFilename string) error {
 	}
 	return nil
 }
+
+func CleanupGCEResources(loadBalancerName string) (err error) {
+	gceCloud, ok := TestContext.CloudConfig.Provider.(*gcecloud.GCECloud)
+	if !ok {
+		return fmt.Errorf("failed to convert CloudConfig.Provider to GCECloud: %#v", TestContext.CloudConfig.Provider)
+	}
+	gceCloud.DeleteFirewall(loadBalancerName)
+	gceCloud.DeleteGlobalForwardingRule(loadBalancerName)
+	gceCloud.DeleteGlobalStaticIP(loadBalancerName)
+	hc, _ := gceCloud.GetHttpHealthCheck(loadBalancerName)
```

Comment:
Yeah it's correct, I assuemd we were doing a health check per taretPort. Girish is verifying the case of one targetPort group failing health checks and another passing, and how that manifests with a single health check per service.